languages/js/src/Host.ts

```diff
@@ -507,12 +507,25 @@ export class Host implements Required<DataFilteringQueryParams> {
       }
       default: {
         let instanceId: number | undefined = undefined;
-        if (isConstructor(v)) instanceId = this.getType(v)?.id;
+        let classId: number | undefined = undefined;
 
         // pass a string class repr *for registered types only*, otherwise pass
         // undefined (allow core to differentiate registered or not)
         const v_cast = v as NullishOrHasConstructor;
         let classRepr: string | undefined = v_cast?.constructor?.name;
+
+        if (isConstructor(v)) {
+          instanceId = this.getType(v)?.id;
+          classId = instanceId;
+        } else {
+          // pass classId for instances of *registered classes* only
+          if (classRepr !== undefined && this.types.has(classRepr)) {
+            classId = this.getType(classRepr)?.id;
+          }
+        }
+
+        // pass classRepr for *registered* classes only, pass undefined
+        // otherwise
```

Comment:
Given the following code:

```js
class A {}
class B extends A {}
oso.registerClass(A);
oso.registerClass(B);
```

should the `classId` & `classRepr` for `B` be those of `A` or `B`?

Reply:
To expand on that a bit, I'm trying to lay out the cases to make sure we're handling them all correctly:

- `v` is a registered class whose superclass is a registered class.
- `v` is a registered class whose superclass isn't a registered class.
- `v` is an instance of a registered class.
- `v` is anything else.

Does that correspond with your understanding of the exhaustive cases?

Reply:
to answer questions in order:
* as `B` is a registered class, `ExternalInstances` of `B` should have discrete `classID` and `classRepr` values to those `ExternalInstance`s representing `A` 

 * in the following V scenarios
   * `v` is registered class whose superclass _is a registered class_
     *  display `classId` and `classRepr` of the registered `v` class, not its superclass
     * `classId` and `instanceId` are identical
   * `v` is a registered class whose superclass _isn't a registered class_
     * display `classId` and `classRepr` of the registered `v` class, not its superclass
     * `classId` and `instanceId` are identical
   * `v` is an instance of a registered class
     *  display `classId` of registered `v` class with unique `instanceId` value
   * `v` is an instance of a non-registered class
     * `classId` is `None` ; `classRepr` is `None`